File: java-grpc-v2/domain/src/main/java/idv/sheng/java_grpc_v2/domain/entity/Greeting.java
package idv.sheng.java_grpc_v2.domain.entity;

import idv.sheng.java_grpc_v2.domain.entity.base.AggregateRoot;
import idv.sheng.java_grpc_v2.domain.valueobj.ServerType;
import lombok.*;

import java.util.UUID;

@Getter
@Setter
public class Greeting extends AggregateRoot<UUID> {
    private String userName;
    private ServerType sender;
    private ServerType recipient;
    private String msg;

    public Greeting() {
        super();
    }

    @Builder
    public Greeting(UUID id, String userName, ServerType sender, ServerType recipient, String msg) {
        this.setId(id);
        this.userName = userName;
        this.sender = sender;
        this.recipient = recipient;
        this.msg = msg;
    }
}

File: python-grpc-v2/python_grpc_v2/domain/entity/greeting.py
import uuid

from python_grpc_v2.domain.base import AggregateRoot
from python_grpc_v2.domain.valueobj.server_type import ServerType


class Greeting(AggregateRoot):
    def __init__(self, id: uuid, user_name: str, sender: int, recipient: int, msg: str):
        super().__init__(id)
        self.user_name = user_name
        self.sender = ServerType.from_code(sender)
        self.recipient = ServerType.from_code(recipient)
        self.msg = msg
